### DISCUSSION TITLE
Why the edge without field doesn't create methods and graphql fields?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2655] [Q&A] [answered]
I noticed that if I use this:

```go
func (Player) Fields() []ent.Field {
  return []ent.Field{
    field.String("name"),

    field.Uint64("team_id").Optional(),
  }
}

func (Player) Edges() []ent.Edge {
  return []ent.Edge{
    edge.To("team", Team.Type).Field("team_id").Unique(),
  }
}
```

it creates in PG the column `team_id`, all the methods like `.SetNillableTeamID()` and all the graphql field predicates in `.graphql` schema (I'm using `entgql`).

If I use the "default" way instead:

```go
func (Player) Fields() []ent.Field {
  return []ent.Field{
    field.String("name"),
  }
}

func (Player) Edges() []ent.Edge {
  return []ent.Edge{
    edge.To("team", Team.Type).Unique(),
  }
}
```

it creates the column (with a different name of course), but doesn't create methods and useful graphql fields.

Why?

**Comment by a8m** [comment, marked as answer]
Hey! 

> If I use the "default" way instead:

There's no "default" way. The edge field is just a way to expose the ID of the edge as a field (mainly for efficiency reasons). If it's not exposed, users can't access it in GraphQL or in the other APIs.
